### PR TITLE
ccache 4.4.1

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

In addition to version bump, this incorporates some changes from #83900 (with some upstream fixes to make test suite work in our situation). The rationale for enabling IPO is described in that PR, as is the reason why running the test suite as part of build process for `ccache` in particular (despite our policy of generally not running checks at build time) makes sense.

### CODE REVIEW

File: Formula/ccache.rb [hunk 1]
```diff
@@ -28,8 +28,14 @@ class Ccache < Formula
...
 
   def install
-    system "cmake", ".", *std_cmake_args
-    system "make", "install"
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, "-DENABLE_IPO=TRUE"
+    system "cmake", "--build", "build"
+
+    # Homebrew compiler shim actively prevents ccache usage (see caveats), which will break the test suite
+    system "cmake", "-E", "env", "CC=/usr/bin/#{DevelopmentTools.default_compiler}",
+           "ctest", "-j#{ENV.make_jobs}", "--test-dir", "build"
```
Comment (carlocab) [inline Formula/ccache.rb:36]:
```suggestion
    with_env(CC: DevelopmentTools.locate(ENV.cc)) do
      system "ctest", "-j#{ENV.make_jobs}", "--test-dir", "build"
    end        
```
I think this should work too and is more consistent with how we override environment variables.

Not sure if using system GCC (`gcc-5`) is the intention on Linux, but this also changes the test to use Homebrew GCC. If we want to keep testing with system GCC, just pass `DevelopmentTools.default_compiler` to `DevelopmentTools.locate`. (Though that is a lot more verbose now...)

Comment (carlocab) [reply Formula/ccache.rb:36]: Or maybe not. I don't remember if we have an audit that rules out calling `ctest`.

**Review by carlocab — DISMISSED** [review]

File: Formula/ccache.rb (hunk 1 above)
Comment (alebcay) [reply Formula/ccache.rb:36]: Ah, I knew there was something like `with_env`! I just didn't know what to search for to find it. Agree that this is more idiomatic Homebrew-wise.

**Review by carlocab — DISMISSED** [review]

File: Formula/ccache.rb [hunk 2]
```diff
@@ -28,8 +28,15 @@ class Ccache < Formula
   fails_with gcc: "5"
 
   def install
-    system "cmake", ".", *std_cmake_args
-    system "make", "install"
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, "-DENABLE_IPO=TRUE"
+    system "cmake", "--build", "build"
+
+    # Homebrew compiler shim actively prevents ccache usage (see caveats), which will break the test suite
```
Comment (carlocab) [inline Formula/ccache.rb:34]: Let's add a comment here explaining why we run `ctest`, in case someone's tempted to remove it in the future.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**